node/get_status_node.go

```diff
@@ -297,23 +294,35 @@ func (n *StatusNode) startWithDB(config *params.NodeConfig) error {
 	if err := n.initServices(config, n.httpServer); err != nil {
 		return err
 	}
-	return n.startGethNode()
-}
 
-// startGethNode starts current StatusNode, will fail if it's already started.
-func (n *StatusNode) startGethNode() error {
-	return n.gethNode.Start()
-}
+	for _, service := range n.services {
+		err = n.registerService(service)
+		if err != nil {
+			name := reflect.TypeOf(service).Name()
+			text := fmt.Sprintf("failed to register service '%s'", name)
+			return errorspkg.Wrap(err, text)
+		}
+	}
 
-func (n *StatusNode) setupRPCClient() (err error) {
-	// setup RPC client
-	gethNodeClient, err := n.gethNode.Attach()
+	err = n.timeSourceSrvc.Start(context.Background())
 	if err != nil {
-		return
+		return errorspkg.Wrap(err, "failed to start time source")
+	}
+
+	for _, service := range n.services {
+		err := service.Start()
```

Comment:
Can this be moved to registration loop so that we don't iterate over services twice?